src/OAuth2Form.tsx

```diff
@@ -6,29 +6,51 @@ import { useEffect, useState, useRef } from "react";
 import { SynapseClient, SynapseConstants } from "synapse-react-client";
 import Login from "synapse-react-client/dist/containers/Login";
 import UserCard from "synapse-react-client/dist/containers/UserCard";
+import { useGetCurrentUserProfile } from "synapse-react-client/dist/utils/hooks/SynapseAPI";
 import { SynapseClientError } from "synapse-react-client/dist/utils/SynapseClientError";
 import { OAuthConsentGrantedResponse } from "synapse-react-client/dist/utils/synapseTypes";
 import { AccessCodeResponse } from "synapse-react-client/dist/utils/synapseTypes/AccessCodeResponse";
 import { OAuthClientPublic } from "synapse-react-client/dist/utils/synapseTypes/OAuthClientPublic";
 import { OIDCAuthorizationRequest } from "synapse-react-client/dist/utils/synapseTypes/OIDCAuthorizationRequest";
 import { OIDCAuthorizationRequestDescription } from "synapse-react-client/dist/utils/synapseTypes/OIDCAuthorizationRequestDescription";
-import { UserProfile } from "synapse-react-client/dist/utils/synapseTypes/UserProfile";
 import { getStateParam, getURLParam, handleErrorRedirect } from "./URLUtils";
 
 export const OAuth2Form = () => {
   const isMounted = useRef(true);
+
   const { accessToken, setAccessToken } = useOAuthAppContext();
-  const [profile, setProfile] = useState<UserProfile>();
+
+  const [error, setError] = useState<any>();
+
+  const onError = (error: Error | OAuthClientError | SynapseClientError) => {
+    debugger;
+    if (error instanceof SynapseClientError && error.status === 401) {
+      // invalid token, so clear it
+      SynapseClient.signOut(() => {
+        setAccessToken(undefined);
+      });
+    } else {
+      handleErrorRedirect(error);
+      setError(error);
+    }
+  };
+
+  // In addition to fetching the current user profile, the success of this request will determine if the current access token is valid.
+  const { data: profile } = useGetCurrentUserProfile({
```

Comment:
Would like to use react-query for more data fetching in this app to clean up lots of logic, but not going to make those changes now.